_overviews/toolkit/sttp-request-body-custom.md

```diff
@@ -9,47 +9,71 @@ next-page: sttp-send-file
 
 {% include markdown.html path="_markdown/install-sttp.md" %}
 
-
 ## The case classes
-In scala, you can represent structures of data using case classes. You could, for example, have a `case class` representing a Person in your program:
+In scala, you can define your objects using case classes.
+You could, for example, have a `case class Person` representing a person in your program:
 ```scala
 case class Person(name: String, surname: String)
 ```
-It has two values inside of it, the `name` and `surname`. 
+An instance of `Person` is defined by two values: its `name` and `surname`. 
 
 ## Custom serializer
-If you wanted to send an object of this type as body, you would need to specify a way to convert this object to a request body. 
-`BodySerializer` comes in handy and allows you to do it once and use it implicitly afterwards.
+
+To send a value of `Person` as the body of a request, you need to define how to transform it into an HTTP body.
+`BodySerializer` comes in handy and allows you to do it once and for all.
 
 ```scala
 import sttp.client3._
-given personSerializer: BodySerializer[Person] = { (p: Person) =>
-    val serialized = s"${p.name},${p.surname}"
-    StringBody(serialized, "UTF-8")
+given BodySerializer[Person] = { (p: Person) =>
+  val serialized = s"${p.name},${p.surname}"
+  StringBody(serialized, "UTF-8")
 }
 ```
-This serializer converts a Person to its name and surname, separated by a comma. 
-`given` keyword may appear strange at first, but it just says that this value may be used later transparently in your code by some functions that needs a `BodySerializer[Person]`. 
-You don't need to think about it for too long, that's the only thing you need to do - as long as this value is available, you will be able to send the Person as a body in a request.
+This serializer converts a `Person` to a `StringBody` by concatenating the `name` and `surname`, separated by a comma. 
+
+The `given` keyword may appear strange at first but is very powerfull.
+It makes the `BodySerializer` available in the current scope, to call the `body` method of an HTTP request and pass it a `Person`.
```

Comment:
> to call the `body` method of an HTTP request and pass it a `Person`.

I don't really understand this part, we don't pass the `BodySerializer`, but allow the user to pass Person as the body.

Reply:
Does this read better?
```
Having the `given BodySerializer[Person]` in the current scope, allows you to call the `body` method of an HTTP request and pass it a `Person`.
```

Reply:
I would rather emphasize the fact that it allows the future method calls to utilize this value. I think the `availablility in the scope` and allowing us to pass a certain type to a method is a bit of a harder explanation to understand. Somebody may wonder how a given changes the method signature and makes it possible to pass a different type. I wanted to go in the direction: We define the Serializer as a given, so any method that wants to use it can do that. And that's what `body` method does, so we can pass Person to it.